executor/habitat_test.go

```diff
@@ -19,23 +24,71 @@ func fakeExecCommand(name string, args ...string) *exec.Cmd {
 }
 
 func TestNewHabitat(t *testing.T) {
-	_, err := NewHabitat(dummyAPICommand(habitatFormat), dummyArgs)
+	_, err := NewHabitat(dummyCommandSpec(habitatFormat), dummyArgs)
 	if err != nil {
 		t.Errorf("err=%q, want nil", err)
 	}
 }
 
+func TestGetPkgDirPath(t *testing.T) {
+	spec := dummyCommandSpec(habitatFormat)
+	hab, _ := NewHabitat(spec, []string{})
+	hab.Store = newDummyStore(validShell, spec, nil)
+	// Note: config.BaseCommandPath is customized for test.
+	// see executor/executor_test.go
+	assert.Equal(t, hab.getPkgDirPath(), filepath.Join(config.BaseCommandPath, "foo-dummy/name-dummy/1.0.1"))
+}
+
+func TestGetPkgFilePath(t *testing.T) {
+	spec := dummyCommandSpec(habitatFormat)
+	hab, _ := NewHabitat(spec, []string{})
+	hab.Store = newDummyStore(validShell, spec, nil)
+	assert.Equal(t, hab.getPkgFilePath(), filepath.Join(config.BaseCommandPath, "foo-dummy/name-dummy/1.0.1/dummy.hart"))
+}
+
+func TestIsDownloaded(t *testing.T) {
+	spec := dummyCommandSpec(habitatFormat)
+	hab, _ := NewHabitat(spec, []string{})
+	hab.Store = newDummyStore(validShell, spec, nil)
+	// Not exists
+	assert.False(t, hab.isDownloaded())
+
+	// 0 size file
+	os.MkdirAll(hab.getPkgDirPath(), 0777)
+	file, _ := os.Create(hab.getPkgFilePath())
+	assert.False(t, hab.isDownloaded())
+
+	// non 0 size file
+	file.Write(([]byte)("dummy script."))
+	assert.True(t, hab.isDownloaded())
+
+	defer file.Close()
+	defer os.RemoveAll(hab.getPkgDirPath())
+}
+
 func TestRunHabitat(t *testing.T) {
 	logBuffer.Reset()
 	command = fakeExecCommand
 	defer func() { command = exec.Command }()
 
-	spec := dummyAPICommand(habitatFormat)
+	spec := dummyCommandSpec(habitatFormat)
+
+	// case remote mode
 	hab, _ := NewHabitat(spec, dummyArgs)
 	err := hab.Run()
 	if err != nil {
 		t.Errorf("err=%q, want nil", err)
 	}
+
+	// case local mode
+	spec.Habitat.Mode = "local"
+	os.Setenv("HABITAT_MODE", "local")
+	hab, _ = NewHabitat(spec, dummyArgs)
+	hab.Store = newDummyStore(validShell, spec, nil)
+	err = hab.Run()
+	if err != nil {
+		t.Errorf("err=%q, want nil", err)
+	}
```

Comment:
I think you need and able to test when failed to get command from store.

Also you can check when the exec got error.

Reply:
ok, I will add those tests.